sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java

```diff
@@ -0,0 +1,620 @@
+            locationType, countryCode, phonePlanId, locationOptionsQueries);
+    }
+
+    Mono<Response<AreaCodes>> getAllAreaCodesWithResponse(
+        String locationType, String countryCode, String phonePlanId, List<LocationOptionsQuery> locationOptions,
+        Context context) {
+        LocationOptionsQueries locationOptionsQueries = new LocationOptionsQueries();
+        locationOptionsQueries.setLocationOptions(locationOptions);
+        return phoneNumberAdminClient.getPhoneNumberAdministrations().getAllAreaCodesWithResponseAsync(
+            locationType, countryCode, phonePlanId, locationOptionsQueries, context);
+    }
+
+    /**
+     * Gets the information for a phone number capabilities update
+     *
+     * @param capabilitiesId ID of the capabilities update.
+     * @return A {@link Mono} containing
+     * a {@link UpdatePhoneNumberCapabilitiesResponse} representing the capabilities update.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<UpdatePhoneNumberCapabilitiesResponse> getCapabilitiesUpdate(String capabilitiesId) {
+        return phoneNumberAdminClient.getPhoneNumberAdministrations().getCapabilitiesUpdateAsync(capabilitiesId);
+    }
+
+    /**
+     * Gets the information for a phone number capabilities update
+     *
+     * @param capabilitiesId ID of the capabilities update.
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#getValue()} value returns
+     * a {@link UpdatePhoneNumberCapabilitiesResponse} representing the capabilities update.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<UpdatePhoneNumberCapabilitiesResponse>> getCapabilitiesUpdateWithResponse(
+        String capabilitiesId) {
+        return phoneNumberAdminClient.getPhoneNumberAdministrations()
+            .getCapabilitiesUpdateWithResponseAsync(capabilitiesId);
+    }
+
+    Mono<Response<UpdatePhoneNumberCapabilitiesResponse>> getCapabilitiesUpdateWithResponse(
+        String capabilitiesId, Context context) {
+        return phoneNumberAdminClient.getPhoneNumberAdministrations()
+            .getCapabilitiesUpdateWithResponseAsync(capabilitiesId, context);
+    }
+
+    /**
+     * Adds or removes phone number capabilities.
+     *
+     * @param phoneNumberCapabilitiesUpdate {@link Map} with the updates to perform
+     * @return A {@link Mono} containing
+     * a {@link UpdatePhoneNumberCapabilitiesResponse} representing the capabilities update.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<UpdateNumberCapabilitiesResponse> updateCapabilities(
+        Map<PhoneNumber, NumberUpdateCapabilities> phoneNumberCapabilitiesUpdate) {
+        Map<String, NumberUpdateCapabilities> capabilitiesMap = new HashMap<>();
+        for (Map.Entry<PhoneNumber, NumberUpdateCapabilities> entry : phoneNumberCapabilitiesUpdate.entrySet()) {
```

Comment:
Add null checks before using user-input (`phoneNumberCapabilitiesUpdate`) and provide helpful error message. Do the same for other methods too.

`Objects.requireNonNull(phoneNumberCapabilitiesUpdate, "'phoneNumberCapabilitiesUpdate' cannot be null");`

Reply:
Perhaps we can update the error handling in a separate PR.